### PR TITLE
fix: optimize reconnection steps.

### PR DESCRIPTION
**Pull request by owny990312**

### CODE REVIEW

File: src/plugins/mqtt_plugin/mqtt_rpc_backend.h [hunk 1]
```diff
@@ -115,6 +120,10 @@ class MqttRpcBackend : public runtime::core::rpc::RpcBackendBase {
 
   std::unique_ptr<runtime::core::util::RpcClientTool<std::shared_ptr<runtime::core::rpc::InvokeWrapper>>>
       client_tool_ptr_;
+
+  std::condition_variable& cv_;
+  std::mutex& cv_mutex_;
+  std::atomic_bool& notified_;
```
Comment (zhangyi1357) [inline src/plugins/mqtt_plugin/mqtt_rpc_backend.h:126]: Take a look at LightSignal in common/utils.

File: src/plugins/mqtt_plugin/mqtt_rpc_backend.h [hunk 2]
```diff
@@ -34,11 +35,12 @@ class MqttRpcBackend : public runtime::core::rpc::RpcBackendBase {
       const std::string& client_id,
       MQTTAsync& client,
       uint32_t max_pkg_size,
-      const std::shared_ptr<MsgHandleRegistry>& msg_handle_registry_ptr)
-      : client_id_(client_id),
```
Comment (AimRT) [inline src/plugins/mqtt_plugin/mqtt_rpc_backend.h:38]: where is it

File: src/plugins/mqtt_plugin/mqtt_channel_backend.h [hunk 3]
```diff
@@ -9,6 +9,7 @@
 #include "core/channel/channel_backend_base.h"
 #include "core/channel/channel_backend_tools.h"
 #include "mqtt_plugin/msg_handle_registry.h"
+#include "util/light_signal.h"
```
Comment (AimRT) [inline src/plugins/mqtt_plugin/mqtt_channel_backend.h:12]: no need

File: src/plugins/mqtt_plugin/mqtt_rpc_backend.h [hunk 4]
```diff
@@ -6,6 +6,7 @@
 #include "core/rpc/rpc_backend_base.h"
 #include "core/util/rpc_client_tool.h"
 #include "mqtt_plugin/msg_handle_registry.h"
+#include "util/light_signal.h"
```
Comment (AimRT) [inline src/plugins/mqtt_plugin/mqtt_rpc_backend.h:9]: delete this line

File: src/plugins/mqtt_plugin/mqtt_channel_backend.h (hunk 3 above)
Comment (AimRT) [reply src/plugins/mqtt_plugin/mqtt_channel_backend.h:12]: delete this line

**Pull request merged**